plugin/magit.vim

```diff
@@ -1116,6 +1116,18 @@ function! magit#stage_hunk(discard)
 	return magit#stage_block(selection, a:discard)
 endfunction
 
+" magit#stage_all
+function! magit#stage_all()
+  call system('git add -u')
+  call magit#update_buffer()
+endfunction
+
+" magit#unstage_all
+function! magit#unstage_all()
+  call system('git reset --mixed')
+  call magit#update_buffer()
+endfunction
+
```

Comment:
The functions are barebones, but they work, assuming the pwd hasn't changed. I don't know who you would do it but this works for me.